django/library/models.py

```diff
@@ -2468,105 +2619,435 @@ def __init__(self, metadata: dict):
+        ]
+
+    @classmethod
+    def to_creative_work(cls, text):
+        return {"@type": "CreativeWork", "text": text}
+
+    @classmethod
+    def convert_authors(cls, common_metadata: CommonMetadata):
+        return [
+            author.contributor.to_codemeta()
+            for author in common_metadata.release_contributor_authors
+        ]
+
+    @classmethod
+    def convert_target_product(cls, common_metadata: CommonMetadata):
         target_product = {
             "@type": "SoftwareApplication",
-            "name": codebase_release.title,
-            "operatingSystem": codebase_release.os,
+            "name": common_metadata.release_title,
+            "operatingSystem": common_metadata.os,
             "applicationCategory": "Computational Model",
         }
-        if codebase_release.live:
+        if common_metadata.live:
             target_product.update(
-                downloadUrl=f"{settings.BASE_URL}{codebase_release.get_download_url()}",
-                releaseNotes=codebase_release.release_notes.raw,
-                softwareVersion=codebase_release.version_number,
-                identifier=codebase_release.permanent_url,
-                sameAs=codebase_release.comses_permanent_url,
+                # FIXME: consider adding a convenience method to generate absolute urls
+                downloadUrl=f"{settings.BASE_URL}{common_metadata.download_url}",
+                releaseNotes=getattr(common_metadata, "release_notes", None),
+                softwareVersion=common_metadata.version,
+                identifier=common_metadata.identifier,
+                sameAs=common_metadata.comses_permanent_url,
             )
-        image_urls = codebase_release.codebase.get_image_urls()
-        if image_urls:
-            target_product.update(screenshot=f"{settings.BASE_URL}{image_urls[0]}")
+        image_url = common_metadata.get_featured_rendition_url
+        if image_url:
+            # FIXME: consider adding a convenience method to generate absolute urls
+            target_product.update(screenshot=f"{settings.BASE_URL}{image_url}")
         return target_product
 
-    @classmethod
-    def convert_keywords(cls, codebase_release: CodebaseRelease):
-        return [tag.name for tag in codebase_release.codebase.tags.all()]
+    def to_json(self):
+        """Returns a JSON string of this codemeta data"""
+        return json.dumps(self.metadata, indent=4)
+
+    def to_dict(self):
+        return self.metadata.copy()
+
+
+class DataCiteMetadata:
+    def __init__(self, metadata: dict):
+        if not metadata:
+            raise ValueError(
+                "Initialize with a base dictionary with DataCite terms mapped to JSON-serializable values"
+            )
+        self.metadata = metadata
 
     @classmethod
-    def convert_platforms(cls, codebase_release: CodebaseRelease):
-        return [tag.name for tag in codebase_release.platform_tags.all()]
+    def from_codebase(cls, codebase: Codebase):
```

Comment:
would it be better if `ReleaseDataciteMetadata` and `CodebaseDataciteMetadata` were 2 separate sibling classes that inherit from the same schema or at least the base fields that are currently duplicated?

Reply:
That sounds like a good way to separate the logic, I'll work on that while continuing to clean up the way their dictionaries are built up